spark/src/main/scala/io/delta/sharing/spark/DeltaSharingSource.scala

```diff
@@ -225,6 +223,55 @@ case class DeltaSharingSource(
     }
   }
 
+  // Function to be called in latestRefreshFunc, to refresh the pre-signed urls in
+  // sortedFetchedFiles. newIdToUrl contains the refreshed urls.
+  private def refreshSortedFetchedFiles(newIdToUrl: Map[String, String]): Unit = {
```

Comment:
We need a lock on this method and all places that modify `sortedFetchedFiles` to avoid race conditions. For example,

- The streaming thread changes it to another `sortedFetchedFiles`.
- The refresh thread changes it back to the old `sortedFetchedFiles`.

Could you also add `@volatile` to `sortedFetchedFiles` so that the streaming thread can see the latest value set by the refresh thread?

Reply:
Will add.
Though it's not necessary as getBatch is a blocking method, so there won't be multiple threads trying to access `sortedFetchedFiles` at the same time.

Reply:
The refresh will be called from the CacheTable as well, but the read happens in the main thread. 
So we should synchronize access to sortedFetchFiles.

Reply:
added.